lib/src/IMiddleware.d.ts

```diff
@@ -0,0 +1,16 @@
+/**
+ * -------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.
+ * See License in the project root for license information.
+ * -------------------------------------------------------------------------------------------
+ */
+import { Context } from "./IContext";
+/**
+ * @interface
+ * @property {Function} execute - The method to execute the middleware
+ * @property {Function} [setNext] - A method to set the next middleware in the chain
+ */
+export interface Middleware {
+    execute: (context: Context) => Promise<any>;
```

Comment:
Execute should return the results of the call back up promise chain.
`execute: (context: Context) => Promise<any>;`

becomes

`execute: (context: Context) => Promise<Context>;`